.github/workflows/wheels.yml

```diff
@@ -128,8 +138,15 @@ jobs:
         env:
           CONFTEST_PATH: ${{ github.workspace }}/conftest.py
           CONFTEST_NAME: conftest.py
+          # Temporary for Python 3.12 testing
+          CIBW_PRERELEASE_PYTHONS: True
+          # TODO: next lines with PIP_ fetch nightly numpy and scipy releases.
+          # They can be removed when there are numpy and scipy releases that
+          # supports Python 3.12
           CIBW_ENVIRONMENT: SKLEARN_SKIP_NETWORK_TESTS=1
                             SKLEARN_BUILD_PARALLEL=3
+                            PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
```

Comment:
Given the [CI issues](https://github.com/scikit-learn/scikit-learn/actions/runs/5784657193/job/15675778145?pr=27027), I think we should set `PIP_EXTRA_INDEX_URL` only for Python 3.12. Otherwise, the other jobs will pick it up and try to build packages from source.

In general, I think it's better to isolate these environment variables to Python 3.12.

Reply:
I would [expand the matirx configuration](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations) for 3.12 with a new `prerelease` item:

```
          - os: macos-latest
            python: 311
            platform_id: macosx_arm64
            prerelease: True
```

and then add it to the `env`:

```
CIBW_PRERELEASE_PYTHONS: ${{ matrix.prerelease }}
```

Then updated `build_wheels.sh` to adjust `CIBW_ENVIRONMENT` based on `CIBW_PRERELEASE_PYTHONS`.